### PR TITLE
Use SafeSerializableSceneObject in TemplateSrv

### PR DESCRIPTION
**Pull request by dprokop**
Complementary work for scenes PR: https://github.com/grafana/scenes/pull/828

fyi @kaydelaney @torkelo @joshhunt @ivanortegaalba

### CODE REVIEW

File: public/app/plugins/datasource/grafana-testdata-datasource/datasource.ts [hunk 1]
```diff
@@ -67,6 +67,9 @@ export class TestDataDataSource extends DataSourceWithBackend<TestDataDataQuery>
     const backendQueries: TestDataDataQuery[] = [];
     const streams: Array<Observable<DataQueryResponse>> = [];
 
+    console.log('Something interpolated...', this.templateSrv.replace('$query0', options.scopedVars));
+    console.log('Something stringified...', JSON.stringify(options));
```
Comment (Sergej-Vlasov) [inline public/app/plugins/datasource/grafana-testdata-datasource/datasource.ts:71]: reminder to remove before merge

**Review by Sergej-Vlasov — APPROVED** [review]
Tested with multiple dashboards and don't see any errors. LGTM 👍

File: public/app/plugins/datasource/grafana-testdata-datasource/datasource.ts (hunk 1 above)
Comment (ivanortegaalba) [inline public/app/plugins/datasource/grafana-testdata-datasource/datasource.ts:71]:
```suggestion
```

File: public/app/plugins/datasource/dashboard/datasource.ts [hunk 2]
```diff
@@ -25,7 +25,9 @@ export class DashboardDatasource extends DataSourceApi<DashboardQuery> {
   }
 
   query(options: DataQueryRequest<DashboardQuery>): Observable<DataQueryResponse> {
-    const scene: SceneObject | undefined = options.scopedVars?.__sceneObject?.value;
+    const scene: SceneObject | undefined = (
+      options.scopedVars?.__sceneObject as SafeSerializableSceneObject
+    )?.valueOf();
```
Comment (ivanortegaalba) [inline public/app/plugins/datasource/dashboard/datasource.ts:30]: Why do we need casting here? :( Can we have a conditional checking the type?

**Review by ivanortegaalba — APPROVED** [review]
I hardly tested against:
✅ Datalinks scoped variables
✅ Repeated panels variables

Captura de pantalla 2024-07-19 a las 13 16 07

Evertything works as expected! Great work Dominik!

**Review by ivanortegaalba — APPROVED** [review]
Looks perfect! LGTM!

File: public/app/features/templating/template_srv.ts [hunk 3]
```diff
@@ -249,8 +249,9 @@ export class TemplateSrv implements BaseTemplateSrv {
   ): string {
     // Scenes compatability (primary method) is via SceneObject inside scopedVars. This way we get a much more accurate "local" scope for the evaluation
     if (scopedVars && scopedVars.__sceneObject) {
+      const sceneObject = (scopedVars.__sceneObject.value as SafeSerializableSceneObject).valueOf();
```
Comment (torkelo) [inline public/app/features/templating/template_srv.ts:252]: @dprokop don't we need a fallback here for older (ie all existing) scene apps that have not upgraded to scenes 
5.6?

**Pull request merged**